### DISCUSSION TITLE
How to serve a static robots.txt or sitemap.xml file from root

### DISCUSSION THREAD

**Discussion by adriaandotcom** [nuxt/nuxt#16658] [Questions] [answered]
Hi there,

I would love to serve a static file from the root of the domain. Useful for `favicon.ico`, `robots.txt`, or `sitemap.xml`.

I just can't find out how to do this. This is my `nuxt.config.ts`:

```js
import { defineNuxtConfig } from "nuxt3";

// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config
export default defineNuxtConfig({
  nitro: {
    // preset: "node-server",
    timing: false,
  },
  build: {
    transpile: ["@heroicons/vue"],
    publicPath: "",
  },
  app: {
    baseURL: "/",
    buildAssetsDir: "/_nuxt/",
    cdnURL: "https://cdn.example.com/",
    head: {
      charset: "utf-8",
      viewport: "width=device-width, initial-scale=1",
    },
  },
  modules: ["@nuxtjs/tailwindcss"],
  compilerOptions: {
    isCustomElement: (tag) => {
      return tag.startsWith("Nuxt");
    },
  },
});
```

I don't want to link them to the `cdnURL`, but to the domain of the Nuxt 3 App itself.

**Comment by benoitdemaegdt** [comment, marked as answer]
Hey 👋 

You can simply create a /public folder and put your files in it.

Example
`/public/favicon.ico`

and then in your `nuxt.config.ts` 
```js
export default defineNuxtConfig({
meta: {
    link: [
      { rel: 'icon', type: 'image/ico', href: '/favicon.ico' },
    ],
}
})
```

**Reply by adriaandotcom** [reply]
You are right, I forgot to copy the `.output/public/_nuxt` folder to my server. Thanks!

**Reply by anjali89r** [reply]
hey @benoitdemaegdt I'm using the SSR and I was able to get the favicon.ico working, but when I try with robots.txt and sitemap.xml it gives me 404 erro

**Reply by benoitdemaegdt** [reply]
@anjali89r I guess you should :
- add `sitemap.xml` inside the public folder
- add `robots.txt` inside the public folder
Then redeploy and you should be able to see it when typing `<YOUR-DOMAIN>/sitemap.xml`

If you are using netlify, try to clear cache and deploy as there might be a caching issue
<img width="486" alt="image" src="https://user-images.githubusercontent.com/25749578/194875270-e2da47ea-5c84-43bf-92e6-2f46b5f01b3a.png">
